lib/Dialect/LWE/IR/LWEOps.td

```diff
@@ -85,7 +85,7 @@ def LWE_EncodeOp : LWE_Op<"encode", [HasEncoding<"output", "encoding", "LWEPlain
   }];
 
   let arguments = (ins
-    SignlessIntegerLike:$plaintext,
+    SignlessIntegerOrFloatLike:$plaintext,
```

Comment:
Perhaps this should come with a verifier that ensures the int/float is compatible with the chosen encoding.

Reply:
Can we make that judgment at this point, though? We could meaningfully want to encode cleartext floats into BGV style plain texts (fixes point representation), pack scalars into SIMD plaintexts, integers into CKKS, etc. I guess the only one that *really* doesn't work is putting a vector into a non-packed encoding - I'll add a verifier for that tomorrow :)

Reply:
Yeah I was thinking mostly that the float input type is only compatible with CKKS. A fixed-point type might be compatible with BGV, but since there's no fixed-point type in SignlessIntegerOrFloatLike (you'd need the quantization dialect I believe), that would default to being an integer type with fixed-point semantics.

Reply:
Done!